core/src/main/java/hudson/scheduler/CronTabList.java

```diff
@@ -78,17 +96,31 @@ public static CronTabList create(String format) throws ANTLRException {
     public static CronTabList create(String format, Hash hash) throws ANTLRException {
         Vector<CronTab> r = new Vector<CronTab>();
         int lineNumber = 0;
+        String timezone = null;
+
         for (String line : format.split("\\r?\\n")) {
             lineNumber++;
             line = line.trim();
+            
+            if(lineNumber == 1 && line.startsWith("TZ=")) {
```

Comment:
Just wondering what's the reason to take `TZ` from the first cron value only?

People trying to run something like this would be confused when the second `TZ` is silently ignored, and the job only runs at London time:
```
TZ=Europe/London    H 8 * * *
TZ=America/New_York H 8 * * *
```

I understand that the idea is to set the timezone once and have it apply to all values, but I think the syntax would be more explicit if the `TZ` was defined on a line of its own, and it was applied to all subsequent lines (until another `TZ` value is found).
But otherwise, at a minimum, I think the form validation should complain if the user enters multiple `TZ` values.

---

Note that the `format` parameter includes the entire crontab text, so this timezone handling will also silently fail if the first line is a comment (i.e. `lineNumber == 1` is no longer true when we check for the `TZ=` prefix):
```
# This job needs to be run in the morning, London time because ...
TZ=Europe/London    H 8 * * *
```